### ISSUE TITLE
Navigation fails with: ERROR [app-router] Error: Unknown object type "htmlanchorelement"

### ISSUE THREAD

**Issue by nenadvicentic** [aurelia/templating-router#71]
**I'm submitting a bug report**

* **Library Version:**
1.3.0

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
8.9.4

* **NPM Version:**
5.6.0

* **Browser:**
Chrome 63 | Firefox 57 | Edge 

* **Language:**
TypeScript 2.6.2

**Current behavior:**
There are 3 child-router sections on the website. After navigating to one specific section (help section, mainly static content) and navigating back to root of the web, navigation to any route fails (main router, child-router, does not matter).

**Expected/desired behavior:**
Navigation should always work, regardless which child-router section was visited.

* ** Error call stack: **

```
ERROR [app-router] Error: Unknown object type "htmlanchorelement"
    at Object._object (http://localhost/scripts/vendor-bundle.js:39016:3181)
    at Object.dispatch (http://localhost/scripts/vendor-bundle.js:39016:2479)
    at http://localhost/scripts/vendor-bundle.js:39016:3095
    at Array.forEach (<anonymous>)
    at Object._object (http://localhost/scripts/vendor-bundle.js:39016:3035)
    at Object.dispatch (http://localhost/scripts/vendor-bundle.js:39016:2479)
    at http://localhost/scripts/vendor-bundle.js:39016:3401
    at Array.forEach (<anonymous>)
    at Object._array (http://localhost/scripts/vendor-bundle.js:39016:3372)
    at Object._object (http://localhost/scripts/vendor-bundle.js:39016:3233)
    at Object.dispatch (http://localhost/scripts/vendor-bundle.js:39016:2479)
    at http://localhost/scripts/vendor-bundle.js:39016:3095
    at Array.forEach (<anonymous>)
    at Object._object (http://localhost/scripts/vendor-bundle.js:39016:3035)
    at Object.dispatch (http://localhost/scripts/vendor-bundle.js:39016:2479)
    at http://localhost/scripts/vendor-bundle.js:39016:3095
From previous event:
    at RouterView.process (http://localhost/scripts/vendor-bundle.js:29930:95)
    at _loop (http://localhost/scripts/vendor-bundle.js:20458:33)
    at NavigationInstruction._commitChanges (http://localhost/scripts/vendor-bundle.js:20474:9)
    at http://localhost/scripts/vendor-bundle.js:20460:71
From previous event:
    at _loop (http://localhost/scripts/vendor-bundle.js:20458:74)
    at NavigationInstruction._commitChanges (http://localhost/scripts/vendor-bundle.js:20474:9)
    at CommitChangesStep.run (http://localhost/scripts/vendor-bundle.js:20310:36)
    at next (http://localhost/scripts/vendor-bundle.js:20279:20)
    at iterate (http://localhost/scripts/vendor-bundle.js:21439:14)
    at inspect (http://localhost/scripts/vendor-bundle.js:21416:16)
    at http://localhost/scripts/vendor-bundle.js:21432:20
    at <anonymous>
From previous event:
    at processPotential (http://localhost/scripts/vendor-bundle.js:21520:35)
    at iterate (http://localhost/scripts/vendor-bundle.js:21431:18)
    at processActivatable (http://localhost/scripts/vendor-bundle.js:21442:12)
    at CanActivateNextStep.run (http://localhost/scripts/vendor-bundle.js:21304:14)
    at next (http://localhost/scripts/vendor-bundle.js:20279:20)
    at http://localhost/scripts/app-bundle.js:387:33
    at <anonymous>
From previous event:
    at http://localhost/scripts/app-bundle.js:380:22
From previous event:
    at AuthorizeStep.run (http://localhost/scripts/app-bundle.js:377:20)
From previous event:
    at LoadRouteStep.run (http://localhost/scripts/vendor-bundle.js:21580:68)
    at next (http://localhost/scripts/vendor-bundle.js:20279:20)
    at iterate (http://localhost/scripts/vendor-bundle.js:21358:14)
    at processDeactivatable (http://localhost/scripts/vendor-bundle.js:21361:12)
    at CanDeactivatePreviousStep.run (http://localhost/scripts/vendor-bundle.js:21292:14)
    at next (http://localhost/scripts/vendor-bundle.js:20279:20)
    at http://localhost/scripts/vendor-bundle.js:20744:16
From previous event:
    at BuildNavigationPlanStep.run (http://localhost/scripts/vendor-bundle.js:20742:58)
    at next (http://localhost/scripts/vendor-bundle.js:20279:20)
    at Pipeline.run (http://localhost/scripts/vendor-bundle.js:20292:14)
    at http://localhost/scripts/vendor-bundle.js:21891:25
From previous event:
    at AppRouter._dequeueInstruction (http://localhost/scripts/vendor-bundle.js:21864:32)
    at http://localhost/scripts/vendor-bundle.js:21855:17
From previous event:
    at AppRouter._queueInstruction (http://localhost/scripts/vendor-bundle.js:21852:14)
    at http://localhost/scripts/vendor-bundle.js:21798:24
    at <anonymous>
From previous event:
    at AppRouter.loadUrl (http://localhost/scripts/vendor-bundle.js:21797:53)
    at BrowserHistory._loadUrl (http://localhost/scripts/vendor-bundle.js:17029:55)
    at BrowserHistory.navigate (http://localhost/scripts/vendor-bundle.js:16974:21)
    at AppRouter.navigate (http://localhost/scripts/vendor-bundle.js:20968:27)
    at Projects.open (http://localhost/scripts/app-bundle.js:2650:26)
    at CallScope.evaluate (http://localhost/scripts/vendor-bundle.js:10759:21)
    at Call.callSource (http://localhost/scripts/vendor-bundle.js:14229:42)
    at DynamicElement.<anonymous> (http://localhost/scripts/vendor-bundle.js:14253:24)
    at CallScope.evaluate (http://localhost/scripts/vendor-bundle.js:10759:21)
    at Listener.callSource (http://localhost/scripts/vendor-bundle.js:14381:42)
    at Listener.handleEvent (http://localhost/scripts/vendor-bundle.js:14390:12)
    at HTMLDocument.handleDelegatedEvent (http://localhost/scripts/vendor-bundle.js:12500:22)
```

**Comment by davismj** [comment]
Sounds like it might be implementation specific. Do you have a repro?

**Comment by nenadvicentic** [comment]
No. I am not able to reproduce the issue in any simple form. I tried to remove all content from problematic child-router section, put single route, etc. As soon as I have (even empty) `configureRouter` method on the class, issue pops back again.

**Comment by davismj** [comment]
@bigopon look familiar to you?

**Comment by nenadvicentic** [comment]
OK, I found the culprit. This happens when I am using `object_hash` npm package. Github repository is here: https://github.com/puleos/object-hash.

What is really strange is that the section that breaks navigation does not use `object_hash`, but once you enter this section navigation breaks apart.

Luckily, on IE11 sections that use actual `object_cache` are throwing error: https://github.com/aurelia/pal-browser/issues/22, so I was able to track issue down.
